pkg/controller/service/servicecontroller.go

```diff
@@ -339,14 +339,23 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 
 		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
 
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, api.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-
-		err := s.createLoadBalancer(service, namespacedName)
+		_, exists, err := s.balancer.GetLoadBalancer(s.loadBalancerName(service), s.zone.Region)
```

Comment:
Perhaps I'm misunderstanding the problem here:
`s.createLoadBalancer` already calls provider's `EnsureLoadBalancer`, and the latter is meant to return success if the loadbalancer already exists.  If that *isn't* happening, that that seems like a straightforward bug in the provider and should be fixed there..

Indeed, I see the openstack cloud provider does not handle the case where the pool already exists - that must have fallen through in the original conversion from "Create" -> "Ensure".  I remember discussing exactly this at the time, but I don't remember the details :/